src/http_server.cc

```diff
@@ -1295,6 +1300,10 @@ void
 HTTPAPIServer::HandleRepositoryIndex(
     evhtp_request_t* req, const std::string& repository_name)
 {
+  if (RestrictedAPI(req, "model-repository")) {
+    return;
```

Comment:
Does only returning from this function is sufficient or we also need to respond the request with error?

Reply:
The function itself sends the response - agree the naming isn't the best.

Could move to a macro: RETURN_AND_RESPOND_IF_RESTRICTED

but that also seemed pretty verbose.

CHECK_RESTRICTED_AND_RESPOND

...